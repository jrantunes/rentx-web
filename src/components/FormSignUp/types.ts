export type UserFormData = {
  name?: string;
  email?: string;
  cnh?: string;
  password?: string;
  confirmPassword?: string;
};
