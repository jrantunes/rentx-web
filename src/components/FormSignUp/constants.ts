import * as yup from "yup";

export const userSchema = yup.object().shape({
  name: yup.string().required("Nome obrigatório!"),
  email: yup
    .string()
    .email("Insira um e-mail válido!")
    .required("E-mail obrigatório!"),
  cnh: yup.string().required("CNH obrigatória!"),
  password: yup
    .string()
    .min(6, "No mínimo 6 caracteres!")
    .required("Senha obrigatória!"),
  confirmPassword: yup
    .string()
    .oneOf([null, yup.ref("password")], "As senhas precisam ser iguais!")
});

export const initialValues = {
  name: "",
  email: "",
  cnh: "",
  password: "",
  confirmPassword: ""
};
