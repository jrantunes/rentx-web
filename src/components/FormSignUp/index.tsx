import {
  AiOutlineCar,
  AiOutlineLock,
  AiOutlineMail,
  AiOutlineUser
} from "react-icons/ai";
import { FormDescription, FormWrapper } from "components/Form";
import TextField from "components/TextField";
import Button from "components/Button";

const FormSignUp = () => {
  return (
    <FormWrapper>
      <FormDescription>
        Faça seu cadastro de forma rápida e fácil.
      </FormDescription>

      <form>
        <TextField
          id="name"
          name="name"
          placeholder="Nome"
          label="Nome"
          labelFor="name"
          icon={<AiOutlineUser />}
        />
        <TextField
          id="email"
          name="email"
          placeholder="E-mail"
          label="E-mail"
          labelFor="email"
          type="email"
          icon={<AiOutlineMail />}
        />
        <TextField
          id="cnh"
          name="cnh"
          placeholder="CNH"
          label="CNH"
          labelFor="cnh"
          icon={<AiOutlineCar />}
        />
        <TextField
          id="password"
          name="password"
          placeholder="Senha"
          label="Senha"
          labelFor="password"
          type="password"
          icon={<AiOutlineLock />}
        />

        <TextField
          id="confirm-password"
          name="confirm-password"
          placeholder="Confirmar senha"
          label="Confirmar senha"
          labelFor="confirm-password"
          type="password"
          icon={<AiOutlineLock />}
        />

        <Button size="small" fullWidth>
          Cadastrar
        </Button>
      </form>
    </FormWrapper>
  );
};

export default FormSignUp;
