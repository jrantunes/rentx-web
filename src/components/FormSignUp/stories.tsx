import FormSignUp from ".";
import { ComponentMeta, ComponentStory } from "@storybook/react";

export default {
  title: "FormSignUp",
  component: FormSignUp
  //args: {}
} as ComponentMeta<typeof FormSignUp>;

export const Default: ComponentStory<typeof FormSignUp> = () => (
  <div style={{ width: 360, margin: "auto" }}>
    <FormSignUp />
  </div>
);
