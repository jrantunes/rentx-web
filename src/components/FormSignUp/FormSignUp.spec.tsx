import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import FormSignUp from ".";

describe("<FormSignUp />", () => {
  it("should render the form", () => {
    const { container } = renderWithTheme(<FormSignUp />);

    expect(screen.getByPlaceholderText(/nome/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/e-mail/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/cnh/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Senha")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Confirmar senha")).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /cadastrar/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the form description", () => {
    renderWithTheme(<FormSignUp />);

    expect(
      screen.getByText(/Faça seu cadastro de forma rápida e fácil/i)
    ).toBeInTheDocument();
  });
});
