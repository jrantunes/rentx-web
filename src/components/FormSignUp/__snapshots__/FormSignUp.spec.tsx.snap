// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render the form 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  padding: 0 1.6rem;
  border: 0.1rem solid #EBEBF0;
  -webkit-transition: border 0.1s ease-in-out;
  transition: border 0.1s ease-in-out;
  cursor: text;
}

.c3:hover,
.c3:focus-within {
  border-color: #AEAEB3;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  border-left: 0.1rem solid #F4F5F6;
  padding-left: 1.6rem;
  margin-left: 1.6rem;
}

.c5 .c7 {
  left: 1.6rem;
}

.c6 {
  font-size: 1.6rem;
  font-weight: 400;
  color: #47474D;
  background: transparent;
  width: 100%;
  padding: 1.6rem 0;
  border: 0;
  outline: none;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c6:focus {
  -webkit-transform: translateY(0.8rem);
  -ms-transform: translateY(0.8rem);
  transform: translateY(0.8rem);
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c6:focus + label {
  opacity: 1;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c6::-webkit-input-placeholder {
  font-size: 1.6rem;
  font-weight: 400;
  color: #AEAEB3;
}

.c6::-moz-placeholder {
  font-size: 1.6rem;
  font-weight: 400;
  color: #AEAEB3;
}

.c6:-ms-input-placeholder {
  font-size: 1.6rem;
  font-weight: 400;
  color: #AEAEB3;
}

.c6::placeholder {
  font-size: 1.6rem;
  font-weight: 400;
  color: #AEAEB3;
}

.c8 {
  cursor: text;
  position: absolute;
  background: transparent;
  color: #AEAEB3;
  font-size: 1.2rem;
  font-weight: 400;
  top: 0.6rem;
  left: 0;
  opacity: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c4 svg {
  width: 2rem;
  height: 2rem;
  color: #7A7A80;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  cursor: pointer;
  font-family: Inter,--apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  font-weight: 500;
  color: #FFFFFF;
  padding: 0.8rem 4rem;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 6rem;
  background-color: #DC1637;
  width: 100%;
}

.c10:hover {
  background-color: #ae112b;
}

.c10:disabled {
  opacity: 0.6;
}

.c10:disabled:hover {
  background-color: #DC1637;
}

.c0 .c2 {
  margin: 0.8rem 0;
}

.c0 .c9 {
  margin: 1.6rem auto 0;
}

.c1 {
  color: #7A7A80;
  font-size: 1.6rem;
  font-weight: 400;
  line-height: 2.4rem;
  margin-bottom: 3.2rem;
  max-width: 24rem;
}

<div
  class="c0"
>
  <p
    class="c1"
  >
    Faça seu cadastro de forma rápida e fácil.
  </p>
  <form>
    <div
      class="c2 "
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </div>
        <div
          class="c5"
        >
          <input
            autocomplete="off"
            class="c6"
            id="name"
            name="name"
            placeholder="Nome"
            value=""
          />
          <label
            aria-hidden="true"
            class="c7 c8"
            for="name"
          >
            Nome
          </label>
        </div>
      </div>
    </div>
    <div
      class="c2 "
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
            />
          </svg>
        </div>
        <div
          class="c5"
        >
          <input
            autocomplete="off"
            class="c6"
            id="email"
            name="email"
            placeholder="E-mail"
            type="email"
            value=""
          />
          <label
            aria-hidden="true"
            class="c7 c8"
            for="email"
          >
            E-mail
          </label>
        </div>
      </div>
    </div>
    <div
      class="c2 "
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M380 704h264c4.4 0 8-3.6 8-8v-84c0-4.4-3.6-8-8-8h-40c-4.4 0-8 3.6-8 8v36H428v-36c0-4.4-3.6-8-8-8h-40c-4.4 0-8 3.6-8 8v84c0 4.4 3.6 8 8 8zm340-123a40 40 0 1 0 80 0 40 40 0 1 0-80 0zm239-167.6L935.3 372a8 8 0 0 0-10.9-2.9l-50.7 29.6-78.3-216.2a63.9 63.9 0 0 0-60.9-44.4H301.2c-34.7 0-65.5 22.4-76.2 55.5l-74.6 205.2-50.8-29.6a8 8 0 0 0-10.9 2.9L65 413.4c-2.2 3.8-.9 8.6 2.9 10.8l60.4 35.2-14.5 40c-1.2 3.2-1.8 6.6-1.8 10v348.2c0 15.7 11.8 28.4 26.3 28.4h67.6c12.3 0 23-9.3 25.6-22.3l7.7-37.7h545.6l7.7 37.7c2.7 13 13.3 22.3 25.6 22.3h67.6c14.5 0 26.3-12.7 26.3-28.4V509.4c0-3.4-.6-6.8-1.8-10l-14.5-40 60.3-35.2a8 8 0 0 0 3-10.8zM840 517v237H184V517l15.6-43h624.8l15.6 43zM292.7 218.1l.5-1.3.4-1.3c1.1-3.3 4.1-5.5 7.6-5.5h427.6l75.4 208H220l72.7-199.9zM224 581a40 40 0 1 0 80 0 40 40 0 1 0-80 0z"
            />
          </svg>
        </div>
        <div
          class="c5"
        >
          <input
            autocomplete="off"
            class="c6"
            id="cnh"
            name="cnh"
            placeholder="CNH"
            value=""
          />
          <label
            aria-hidden="true"
            class="c7 c8"
            for="cnh"
          >
            CNH
          </label>
        </div>
      </div>
    </div>
    <div
      class="c2 "
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
            />
          </svg>
        </div>
        <div
          class="c5"
        >
          <input
            autocomplete="off"
            class="c6"
            id="password"
            name="password"
            placeholder="Senha"
            type="password"
            value=""
          />
          <label
            aria-hidden="true"
            class="c7 c8"
            for="password"
          >
            Senha
          </label>
        </div>
      </div>
    </div>
    <div
      class="c2 "
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
            />
          </svg>
        </div>
        <div
          class="c5"
        >
          <input
            autocomplete="off"
            class="c6"
            id="confirm-password"
            name="confirm-password"
            placeholder="Confirmar senha"
            type="password"
            value=""
          />
          <label
            aria-hidden="true"
            class="c7 c8"
            for="confirm-password"
          >
            Confirmar senha
          </label>
        </div>
      </div>
    </div>
    <button
      class="c9 c10"
      color="red"
    >
      <span>
        Cadastrar
      </span>
    </button>
  </form>
</div>
`;
